addons/account/models/account_move.py

```diff
@@ -2062,7 +2062,7 @@ def post(self):
             # When the accounting date is prior to the tax lock date, move it automatically to the next available date.
             # /!\ 'check_move_validity' must be there since the dynamic lines will be recomputed outside the 'onchange'
             # environment.
-            if move.company_id.tax_lock_date and move.date <= move.company_id.tax_lock_date:
+            if (move.company_id.tax_lock_date and move.date <= move.company_id.tax_lock_date) and move.line_ids.tax_ids:
```

Comment:
To be complete (in case some tags were set manually on the move lines without giving any tags):

... and (move.line_ids.tax_ids or move.line_ids.tag_ids)

Reply:
ok, updated, thank you for the suggestion